ci-operator/step-registry/platform-external/cluster/aws/install/platform-external-cluster-aws-install-commands.sh

```diff
@@ -0,0 +1,421 @@
+# TODO(mtulio): move to CloudFormation StackSet (single stack deployment)
+TEMPLATES_BASE=https://raw.githubusercontent.com/mtulio/installer
+TEMPLATES_VERSION=upi-installer-ci
+TEMPLATES_PATH=upi/aws/cloudformation
+
+TEMPLATE_URL=${TEMPLATES_BASE}/${TEMPLATES_VERSION}/${TEMPLATES_PATH}
+TEMPLATES=( "01_vpc.yaml" )
+TEMPLATES+=( "02_cluster_infra.yaml" )
+TEMPLATES+=( "03_cluster_security.yaml" )
+TEMPLATES+=( "04_cluster_bootstrap.yaml" )
+TEMPLATES+=( "05_cluster_master_nodes.yaml" )
+TEMPLATES+=( "06_cluster_worker_node.yaml" )
+
+#TEMPLTE_DEST="/var/lib/upi-installer/${TEMPLATES_PATH}"
+TEMPLTE_DEST="/tmp"
+
+for TEMPLATE in "${TEMPLATES[@]}"; do
+  log "Updating ${TEMPLATE}"
+  curl -sL "${TEMPLATE_URL}/${TEMPLATE}" > "${TEMPLTE_DEST}/${TEMPLATE}"
+done
+
+echo "================================="
+echo "CREATING INFRASTRUCTURE RESOURCES"
+echo "================================="
+
+# The oc binary is placed in the shared-tmp by the test container and we want to use
+# that oc for all actions.
+
+GATHER_BOOTSTRAP_ARGS=
+NEW_STACKS="${SHARED_DIR}/aws_cfn_stacks"
+# mkdir -vp $NEW_STACKS
+touch $NEW_STACKS
+
+function add_param_to_json() {
+    local k="$1"
+    local v="$2"
+    local param_json="$3"
+    if [ ! -e "$param_json" ]; then
+        echo -n '[]' > "$param_json"
+    fi
+    cat <<< "$(jq  --arg k "$k" --arg v "$v" '. += [{"ParameterKey":$k, "ParameterValue":$v}]' "$param_json")" > "$param_json"
+}
+
+function gather_bootstrap_and_fail() {
+  if test -n "${GATHER_BOOTSTRAP_ARGS}"; then
+    openshift-install --dir=${SHARED_DIR} gather bootstrap --key "${SSH_PRIV_KEY_PATH}" ${GATHER_BOOTSTRAP_ARGS}
+  fi
+
+  return 1
+}
+
+# echo "Installing from initial release ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}"
+OPENSHIFT_INSTALL_INVOKER=openshift-internal-ci/${JOB_NAME}/${BUILD_ID}
+AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/.awscred
+EXPIRATION_DATE=$(date -d '4 hours' --iso=minutes --utc)
+# PULL_SECRET=${CLUSTER_PROFILE_DIR}/pull-secret
+
+export OPENSHIFT_INSTALL_INVOKER
+export AWS_SHARED_CREDENTIALS_FILE
+export EXPIRATION_DATE
+
+base_domain=$(yq3 r "${SHARED_DIR}/install-config.yaml" 'baseDomain')
+CLUSTER_NAME=$(yq3 r "${SHARED_DIR}/install-config.yaml" 'metadata.name')
+
+echo ${AWS_REGION} > ${SHARED_DIR}/AWS_REGION
+echo ${CLUSTER_NAME} > ${SHARED_DIR}/CLUSTER_NAME
+
+# TODO(mtulio): move to step vars
+MACHINE_CIDR=10.0.0.0/16
+
+# begin bootstrapping
+if openshift-install coreos print-stream-json 2> "${ARTIFACT_DIR}/err.txt" > /tmp/coreos.json; then
+  RHCOS_AMI="$(jq -r --arg region "$AWS_REGION" '.architectures.x86_64.images.aws.regions[$region].image' /tmp/coreos.json)"
+  # if [[ "${CLUSTER_TYPE}" == "aws-arm64" ]] || [[ "${OCP_ARCH}" == "arm64" ]]; then
+  #   RHCOS_AMI="$(jq -r --arg region "$AWS_REGION" '.architectures.aarch64.images.aws.regions[$region].image' coreos.json)"
+  # fi
+else
+  RHCOS_AMI="$(jq -r --arg region "$AWS_REGION" '.amis[$region].hvm' /var/lib/openshift-install/rhcos.json)"
+fi
```

Comment:
this can be moved to pre-steps as a long-term goal to prevent using `openshift-install` in installing for provider-specific steps.